src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -306,6 +368,12 @@ public Task<TResource> Replace(string databaseName, Task<TResource> task)
                             _parent._caseSensitive.TryRemove(mapping, out Task<TResource> _);
                         }
                     }
+
+                    // We need to refresh all of them.
+                    _parent._readonlyCaseSensitive = _parent._caseSensitive.ToFrozenDictionary();
+                    _parent._readonlyCaseInsensitive = _parent._caseInsensitive.ToFrozenDictionary();
+                    _parent._readonlyResourceDetails = _parent._resourceDetails.ToFrozenDictionary();
+
```

Comment:
Shouldn't this be also done in the two steps method?
For that matter, can we wrap that in a method to ensure that this is not duplicated?

Reply:
@redknightlois ?

Reply:
What do you mean by the two-steps method? In this particular code, the update will be executed under the lock every time, regardless of what happens.

Reply:
Not a big deal, but yes. Fixed.